code/modules/projectiles/gun_system.dm

```diff
@@ -1594,7 +1594,7 @@
 		return 0
 
 	switch(gun_firemode)
```

Comment:
You're ignoring the deployed scatter change + there is no reason to have a switch here